lib/plugins/cleaner.rb

```diff
@@ -0,0 +1,47 @@
+require 'rubygems/package'
+require_relative '../moonshot/creds_helper'
+
+module Moonshot
+  module Plugins
+    # Moonshot plugin class for removing artifacts when a stage gets deleted.
+    class Cleaner # rubocop:disable Metrics/ClassLength
+      include Moonshot::CredsHelper
+
+      def initialize(bucket_name)
+        @bucket_name = bucket_name
+      end
+
+      def pre_delete(resources)
```

Comment:
I wonder if pre_delete is what we want here. The artifacts are in use before we delete the stack. Perhaps post_delete is a better fit?